tests/test_ui.py

```
@@ -110,127 +102,98 @@ def test_results_renderer_single_card_with_members():
         ],
     }
 
-    mock_state = MagicMock()
-    mock_state.get.return_value = 2
+    mock_state = DummySessionState({"num_groups_target": 2})
+
+    def mock_cols(n, **_):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `mock_cols`

(ANN202)